File: Java/heap/KLargestElements.java
package heap;

public class KLargestElements {
    public static void kLargestElements(int[] arr, int k) {
        PriorityQueue pq = new PriorityQueue();
        for (int i = 0; i < k; i++)
            pq.add(arr[i]);
        for (int i = k; i < arr.length; i++) {
            if (arr[i] > pq.peek()) {
                pq.poll();
                pq.add(arr[i]);
            }
        }
        while (!pq.isEmpty())
            System.out.print(pq.poll() + " ");
        System.out.println();
    }

    public static void main(String[] args) {
        int[] arr = { 5, 15, 10, 20, 8, 25, 18 };
        kLargestElements(arr, 3);
    }
}

File: Python/heap/k_largest_elements.py
import priority_queue as pq


def k_largest_elements(arr, k):
    for i in range(k):
        pq.add(arr[i])
    for i in range(k, len(arr)):
        if arr[i] > pq.peek():
            pq.poll()
            pq.add(arr[i])
    while not pq.is_empty():
        print(pq.poll(), end=" ")
    print()


def main():
    arr = [5, 15, 10, 20, 8, 25, 18]
    k_largest_elements(arr, 3)


if __name__ == "__main__":
    main()
